File: Packet/MapSeed.cs
﻿using System.IO;

namespace Resources.Packet {
    public class MapSeed : Packet{
        public int seed;

        public MapSeed() : base(PacketID.MapSeed) { }
        public MapSeed(BinaryReader reader) : this() {
            seed = reader.ReadInt32();
        }

        protected override void WritePacketData(BinaryWriter writer) {
            writer.Write(seed);
        }
    }
}

File: src/main/kotlin/packets/MapSeed.kt
package com.github.lastexceed.cubeworldnetworking.packets

import com.github.lastexceed.cubeworldnetworking.utils.*

class MapSeed(
	val seed: Int
) : Packet(PacketId.MapSeed) {
	override suspend fun writeTo(writer: Writer) {
		writer.writeInt(seed)
	}

	companion object : CwDeserializer<MapSeed> {
		override suspend fun readFrom(reader: Reader) = MapSeed(reader.readInt())
	}
}